src/users/users.repository.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable, NotFoundException } from '@nestjs/common'
+import { Prisma } from '@prisma/client'
+import { PrismaService } from 'src/prisma/prisma.service'
+import { UserEntity } from 'src/users/entities/user.entity'
+import calculatePagination from 'src/utils/calculatePagination'
+
+@Injectable()
+export class UsersRepository {
+    constructor(private prisma: PrismaService) {}
+
+    async getUsers(
+        page: number,
+        pageSize: number,
+        filter: Prisma.UsersWhereInput,
+    ): Promise<UserEntity[]> {
+        const users = await this.prisma.users.findMany({
+            where: filter,
+            take: pageSize,
+            skip: calculatePagination(page, pageSize),
+            include: {
+                Student: true,
+            },
+        })
+
+        return users
+    }
+
+    async getUserById(id: number, returnStudent: boolean): Promise<UserEntity> {
+        const user = await this.prisma.users.findUnique({
```

Comment:
I prefer findOrThorw

Reply:
findOrThrow it need to handle error in `http.exception.filter` I do some research but I dont find way to handle prisma not found exception if you have interesting resource let me know

Reply:
Ok, I got it.
I will research another way to handle error